### DISCUSSION TITLE
Disable html optimizer

### DISCUSSION THREAD

**Discussion by eirikb** [parcel-bundler/parcel#5910] [Q&A] [answered]
Hi

Is it possible to completely disable optimizing of html?
I'm trying to migrate an ancient AngularJS app to Parcel, which seems to work very well, except for some issues with optimizing of the template files. I need the templates as pure inline string.
I still want to optimize everything else, so `--no-minify` is not an option.

My current **.parcelrc** looks like this:
```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.html": [
      "@parcel/transformer-inline-string"
    ]
  }
}
```
This way I didn't have to change all `require` to `bundle-text:`.

The issue I have is this template (snippet from larger template):
```html
<i class="fa {{module.icon.icon}} {{ module.icon.color }}"></i>
```
ending up like this:
```html
<i class="{{ {{module.icon.icon}} }} fa module.icon.color"></i>
```

Simply disabling any optimization would solve the problem.  

I'm aware of this issue: https://github.com/parcel-bundler/parcel/issues/2561 and that it goes through `htmlnano` (at least after setting `optimizers` to that in **.parcelrc**). 
I've tried running **htmlnano** manually, and all presets (and the example config from the issue) produce the same output:
```js
const htmlnano = require('htmlnano');
for (const preset of Object.values(htmlnano.presets)) {
  htmlnano.process('<i class="fa {{module.icon.icon}} {{ module.icon.color }}"></i>', {}, preset).then(r => console.log(r.html))
}
```

Maybe this could be considered a bug in `htmlnano`, or I'm doing something very wrong, but complete disable would work.

I've noticed that removing space between `{{` and `}}` seems to solve the issue, in a manner, but I'd really not like to fix this manually as not all the code is even mine, and there would be pitfalls (and other devs could introduce  bugs in new changes).

**Comment by mischnic** [comment, marked as answer]
> completely disable optimizing of html

```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.html": [
      "@parcel/transformer-inline-string"
    ]
  },
  "optimizers": {
    "*.html": []
  }
}
```

**Reply by eirikb** [reply]
You can't imagine how dumb I feel right now.  

Thanks a bunch, that works perfectly!

**Reply by eirikb** [reply]
Just curious, would it be possible to say ignore all `*.html`-files under a specific folder, but include everything else?
All my AngularJS templates are located under **src/tpl/**, but later I might need a **index.html** to be processed as usual.

I tried using ` "**/tpl/**/*.html"` as glob pattern for transformer, but I'm not sure what to use for the optimizer because it is against bundled path.

This is my latest test (does not work):
```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "**/tpl/**/*.html": [
      "@parcel/transformer-inline-string"
    ]
  },
  "optimizers": {
    "*.html": [],
    "index.html": ["@parcel/optimizer-htmlnano"]
  }
}
```

**Reply by mischnic** [reply]
Maybe `"dist/index.html": ["@parcel/optimizer-htmlnano"]` instead then? (Or whatever your dist folder is called)

**Reply by eirikb** [reply]
That makes sense, but didn't work, however `**/index.html` did!
This is really cool.

For anyone randomly stumbling upon this thread, this is the final config:
```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "**/tpl/**/*.html": [
      "@parcel/transformer-inline-string"
    ]
  },
  "optimizers": {
    "**/index.html": ["@parcel/optimizer-htmlnano"],
    "*.html": []
  }
}
```
